### PR TITLE
Enable more Ruff PYI rules with autofixes

### PR DESCRIPTION
**Pull request by Avasam**
Added rules: `PYI014`, `PYI015`, `PYI030`, `PYI036`, `PYI044`, `PYI062`

I originally split this into 3 sections (`Fix is always available.`, `Fix is sometimes available.`, and `We sometimes noqa these`), but I'm not sure if that's something worth doing, keeping it here for archiving purposes:
```toml
select = [
    ...

	# PYI rules that sometimes autofix, but we should always be able to manually fix,
	# so there's still no conflict/duplicate with flake8-pyi
	"PYI013", # Non-empty class bodies must not contain `...`
	"PYI016", # Duplicate union member `{}`
	"PYI025", # Use `from collections.abc import Set as AbstractSet` to avoid confusion with the `set` builtin
	"PYI030", # Multiple literal members in a union. Use a single literal, e.g. `Literal[{}]`
	"PYI036", # Star-args in `{method_name}` should be annotated with `object`
	"PYI044", # `from __future__ import annotations` has no effect in stub files, since type checkers automatically treat stubs as having those semantics
	"PYI055", # Multiple `type[T]` usages in a union. Combine them into one, e.g., `type[{union_str}]`.
	"PYI062", # Duplicate literal member `{}`
	# Rules that have autofixes, but we sometimes disable on a per-line basis
	# "PYI026", Waiting for this mypy bug to be fixed: https://github.com/python/mypy/issues/16581
	# "PYI029", __str__ and __str__ sometimes have to be used for classes other than `object`
	# Sometimes we still wanna be explicit about a default value nonetheless
	# "PYI011",
	# "PYI053", Also removes `Literal[{string_too_long}]` from an annotation
	# "PYI054",
]
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,16 +48,31 @@ select = [
...
+    "PYI036", # Star-args in `{method_name}` should be annotated with `object`
+    "PYI044", # `from __future__ import annotations` has no effect in stub files, since type checkers automatically treat stubs as having those semantics
+    "PYI055", # Multiple `type[T]` usages in a union. Combine them into one, e.g., `type[{union_str}]`.
+    "PYI062", # Duplicate literal member `{}`
+    # Rules that have autofixes, but we sometimes disable on a per-line basis
+    # "PYI026", Waiting for this mypy bug to be fixed: https://github.com/python/mypy/issues/16581
+    # "PYI029", __str__ and __str__ sometimes have to be used for classes other than `object`
+    # Sometimes we still wanna be explicit about a default value nonetheless
+    # "PYI011", Also has false-positives with `_typeshed.sentinel`
+    # "PYI053", Also removes `Literal[{string_too_long}]` from an annotation
```
Comment (Avasam) [inline pyproject.toml]: We noqa that one twice in nanoid stubs for the paramether `alphabet: str = "_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"`

Ruff issue for the bad autofix in `Literal`: https://github.com/astral-sh/ruff/issues/12995

File: pyproject.toml [hunk 2]
```diff
@@ -46,18 +46,26 @@ select = [
     "E", # pycodestyle Error
     "F", # Pyflakes
     "W", # pycodestyle Warning
-    # PYI: only enable rules that always autofix, avoids duplicate # noqa with flake8-pyi
+    # PYI: only enable rules that have autofixes and that we always want to fix (even manually),
+    # avoids duplicate # noqa with flake8-pyi and flake8-noqa flagging `PYI` codes
     # See https://github.com/plinss/flake8-noqa/issues/22
```
Comment (Avasam) [inline pyproject.toml:51]: If anyone has a contact to @plinss , that's be a nice flake8-noqa feature to get merged ^^"

**Review by JelleZijlstra — APPROVED** [review]

**Comment by Avasam** [comment]
Doesn't look like Ruff 0.6.2 added anything that should affect this (`PYI053` has less false-positives, but it's still disabled on our side)

**Comment by Avasam** [comment]
Jelle approved 2 weeks ago and it's unchallenged. I'll merge it. If you see an issue, open a new PR!

**Pull request merged**